R/format_PEW_temp.R

```diff
@@ -0,0 +1,541 @@
+                      # I do not use them.
+                      AvgChickMass = mean(as.numeric(Mass), na.rm = TRUE),
+                      NumberChicksMass = sum(!is.na(Mass)),
+                      AvgTarsus = mean(as.numeric(Tarsus), na.rm = TRUE),
+                      NumberChicksTarsus = sum(!is.na(Tarsus)),
+                      OriginalTarsusMethod = NA_character_) %>%
+    ungroup() %>%
+    dplyr::mutate(AvgTarsus = ifelse(AvgTarsus == "NaN", NA, AvgTarsus),
+                  AvgChickMass = ifelse(AvgChickMass == "NaN", NA, AvgChickMass),
+                  NumberChicksMass = ifelse(NumberChicksMass == 0, NA, NumberChicksMass),
+                  NumberChicksTarsus = ifelse(NumberChicksTarsus == 0, NA, NumberChicksTarsus))
+
+
+  #### Join parents and chick data
+  Brood_data <-
+    parents_brood_data_2 %>%
+    left_join(chicks_measurements, by = "BroodID") %>%
+    #### Rename
+    dplyr::rename(LayDate_observed = LayDate,
+                  NumberFledged_observed = NumberFledged) %>%
+    #### Final arrangement
+    dplyr::select(BroodID, PopID, BreedingSeason, Species, Plot, LocationID,
+                  FemaleID, MaleID,
+                  ClutchType_observed, ClutchType_calculated,
+                  LayDate_observed, LayDate_min, LayDate_max,
+                  ClutchSize_observed, ClutchSize_min, ClutchSize_max,
+                  HatchDate_observed, HatchDate_min, HatchDate_max,
+                  BroodSize_observed, BroodSize_min, BroodSize_max,
+                  FledgeDate_observed, FledgeDate_min, FledgeDate_max,
+                  NumberFledged_observed, NumberFledged_min, NumberFledged_max,
+                  AvgEggMass, NumberEggs, AvgChickMass, NumberChicksMass,
+                  AvgTarsus, NumberChicksTarsus, OriginalTarsusMethod, ExperimentID)
+
+  return(Brood_data)
+
+}
+
+create_capture_PEW <- function(pew_data, Brood_data) {
+
+  Brood_data_sel <-
+    Brood_data %>%
+    dplyr::select(BreedingSeason, Species, PopID, BroodID, LocationID,
+                  HatchDate_observed)
+
+
+  Capture_data_temp <-
+    pew_data %>%
+    #### Rename variables
+    dplyr::rename(LocationID = NestboxID) %>%
+    dplyr::select(-DateRingingChicks) %>%
+    #### Create new variables
+    dplyr::mutate(#### ASSUMPTION: There is no captureID info, so I made this up
+                  CaptureID = paste(IndvID, LocationID,
+                                    paste0(lubridate::year(CaptureDate),
+                                           sprintf("%02d", lubridate::month(CaptureDate)),
+                                           sprintf("%02d", lubridate::day(CaptureDate))),
+                                    sep = "_"),
+      #### ASSUMPTION: Some times are strange, to simplify, round to whole hour
+      #### ? do they correspond only to the experimental observations? keep/remove?
+      CaptureTime = ifelse(is.na(ObservationTimeH),
+                           NA,
```

Comment:
Better to specify. If we try to run the pipeline on a subset of the data where ObservationTimeH is always missing this will return a logical NA rather than a character (which doesn't fit our standard protocol).

```suggestion
                           NA_character_,
```

Reply:
Right!